__tests__/app.test.js

```diff
@@ -118,7 +118,89 @@ describe("GET /api/articles", () => {
         expect(datedArticles).toBeSortedBy("created_at", { descending: true });
       });
   });
-});
+  test("200: The articles can be queried to be sorted by any valid column, default created_at and order descending", () => {
+    return request(app)
+      .get("/api/articles?sort_by=article_id")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("article_id", { descending: true });
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles?sort_by=votes")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toBeSortedBy("votes", { descending: true });
+          });
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles?sort_by=author")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toBeSortedBy("author", { descending: true });
+          });
+      });
+  });
+  test("200: The articles can be queried to order by ascending or descending", () => {
+    return request(app)
+      .get("/api/articles?order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        const datedArticles = articles.map((article) => {
+          article.created_at = Date(article.created_at);
+          return article;
+        });
+        expect(datedArticles).toBeSortedBy("created_at");
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles?order")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            const datedArticles = articles.map((article) => {
+              article.created_at = Date(article.created_at);
+              return article;
+            });
+            expect(datedArticles).toBeSortedBy("created_at", {
+              descending: true,
+            });
+          });
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles?order=asc&&sort_by=votes")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toBeSortedBy("votes");
+          });
+      });
+  });
+  test("400: Gives bad request when trying to sort by an invalid column", () => {
+    return request(app)
+      .get("/api/articles?sort_by=name")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+  test("400: Gives bad request when trying to order by an invalid parameter", () => {
+    return request(app)
+      .get("/api/articles?order=bob")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+  test("400: Gives bad request when query is not valid", () => {
+    return request(app)
+      .get("/api/articles?dsfkj=sdfj")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
```

Comment:
This isnt a behaviour that should throw an err, but something that should just be ignored